packages/modular-scripts/src/__tests__/index.test.ts

```diff
@@ -37,113 +33,13 @@ function modular(str: string, opts: Record<string, unknown> = {}) {
-  appPath: string,
-  opts: Record<string, unknown> = {},
-): Promise<DevServer> {
-  const devServer = modular(`start ${appPath}`, {
-    cleanup: true,
-    ...opts,
-  });
-
-  await new Promise((resolve, reject) => {
-    if (!devServer.stdout) {
-      return reject(
-        new Error(
-          'The dev server could not produce any output on /dev/stdout.',
-        ),
-      );
-    }
-
-    if (!devServer.stderr) {
-      return reject(
-        new Error(
-          'The dev server could not produce any output on /dev/stderr.',
-        ),
-      );
-    }
-
-    // If the Promise has completed we need to bail out of any further handlers that
-    // are executed, because otherwise we can end up trying to log after tests are done.
-    // We use a 'completed' variable to do this -- in an ideal world we'd cleanup
-    // the listeners.
-    //
-    // See: https://github.com/jpmorganchase/modular/pull/107#discussion_r493791918
-    let completed = false;
-
-    const startAppTimeout = setTimeout(() => {
-      if (completed) return;
-
-      completed = true;
-      reject(
-        new Error(
-          `The app at ${appPath} never started within the configured ${START_APP_TIMEOUT}ms timeout period.`,
-        ),
-      );
-    }, START_APP_TIMEOUT);
-
-    devServer.stdout.on('data', (data: Buffer) => {
-      if (completed) return;
-
-      const output = data.toString();
-      if (/Something is already running on port (\d+)./.test(output)) {
-        clearTimeout(startAppTimeout);
-
-        completed = true;
-        return reject(new Error(output));
-      }
-      if (/Compiled successfully!/.test(output)) {
-        clearTimeout(startAppTimeout);
-
-        completed = true;
-        return resolve(true);
-      }
-    });
-
-    devServer.stderr.on('data', (data: Buffer) => {
-      if (completed) return;
-
-      const output = data.toString();
-
-      logger.error(output);
-
-      clearTimeout(startAppTimeout);
-
-      completed = true;
-      return reject(new Error(output));
-    });
-
-    // eslint-disable-next-line @typescript-eslint/no-floating-promises
-    devServer.on('error', (err: Error) => {
-      if (completed) return;
-
-      logger.error(err);
-
-      clearTimeout(startAppTimeout);
-
-      completed = true;
-      reject(err);
-    });
-  });
-
-  // We can't return the raw execa child process,
-  // because we need to `await` until the dev server
-  // starts above, but don't want to wait until the
-  // process itself finishes.
-  return {
-    kill: () => {
-      devServer.kill('SIGKILL');
-      return devServer;
-    },
-  };
-}
-
 function cleanup() {
-  rimraf.sync(path.join(packagesPath, 'sample-app'));
   rimraf.sync(path.join(packagesPath, 'sample-view'));
   rimraf.sync(path.join(packagesPath, 'sample-package'));
   rimraf.sync(path.join(packagesPath, 'nested'));
-  rimraf.sync(path.join(modularRoot, 'dist'));
+  rimraf.sync(path.join(modularRoot, 'dist/sample-view'));
```

Comment:
why not just delete the dist folder?

Reply:
Because `dist/sample-app` still needs to exist if the tests were to be running at the same time.

Reply:
We set the tests to run in band sequentially. If this is depending on another test file or another test file is depending on this test file to have a built app, then we should fix that.

Reply:
In some sense it's not - `/dist` existing is fine because that's our output folder as well.What we really need to do is to fix the tests to run in a directory which is not the the repo!